sentry-sidekiq/spec/sentry/sidekiq/sentry_context_middleware_spec.rb

```diff
@@ -0,0 +1,84 @@
+require "spec_helper"
+
+class TestWorker
+  include Sidekiq::Worker
+
+  def perform; end
+end
+
+RSpec.shared_context "sidekiq", shared_context: :metadata do
+  let(:client) do
+    Sidekiq::Client.new.tap do |client|
+      client.middleware do |chain|
+        chain.add Sentry::Sidekiq::SentryContextClientMiddleware
+      end
+    end
+  end
+
+  let(:random_empty_queue) do
+    Sidekiq::Queue.new(rand(10_000)).tap do |queue|
+      queue.clear
+    end
+  end
+end
+
+RSpec.describe Sentry::Sidekiq::SentryContextServerMiddleware do
+  include_context "sidekiq"
+
+  before { perform_basic_setup }
+
+  after do
+    Sidekiq.server_middleware do |chain|
+      # Remove the middleware with testing argument and re-add the middleware
+      # to restore the chain.
+      chain.remove described_class
+      chain.add described_class
+    end
+  end
+
+  it "sets user to the current scope from the job" do
+    user = { "id" => rand(10_000) }
+    Sentry.set_user(user)
+
+    Sidekiq.server_middleware do |chain|
+      chain.add described_class, testing_only_callback: lambda { |scope|
+        expect(scope.user).to eq(user)
+      }
+    end
+
+    queue = random_empty_queue
+    options = { fetch: Sidekiq::BasicFetch.new(queues: [queue.name]) }
+    processor = Sidekiq::Processor.new(nil, options)
+
+    client.push('queue' => queue.name, 'class' => TestWorker, 'args' => [])
+
+    expect(queue.size).to be(1)
+    processor.process_one
+    expect(queue.size).to be(0)
```

Comment:
it's a bit tricky but you can test user setting with transaction events:

1. Enable tracing

```ruby
perform_basic_setup do |config|
  config.traces_sample_rate = 1.0
end
```

2. process the job (`processor.process_one`)
3. use `Sentry.get_current_client.transport.events.first` to get the event.
4. check if the event contains the correct user data.

besides using tracing events, I think making `TestWorker` raise exception should also generate an event that contains the user info. personally I'd test both scenarios to make sure the user data is properly set under any condition.